sdk/core/Azure.Core/tests/RetryPolicyTestBase.cs

```diff
@@ -62,6 +64,71 @@ public async Task OnlyRetriesRetriableCodes()
             Assert.AreEqual(501, response.Status);
         }
 
+        [Test]
+        public async Task ShouldRetryIsCalledOnlyForErrors()
+        {
+            (HttpPipelinePolicy policy, AsyncGate<TimeSpan, object> gate) = CreateRetryPolicy(maxRetries: 3);
```

Comment:
consider factoring duplicate setup code into a helper method just to clarify what is different about each test scenario.